<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>qSlicerTablesModuleWidget</class>
 <widget class="qSlicerWidget" name="qSlicerTablesModuleWidget">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>467</width>
    <height>502</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Minimum" vsizetype="Preferred">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="windowTitle">
   <string>Tables</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../qSlicerTablesModule.qrc">
    <normaloff>:/Icons/Tables.png</normaloff>:/Icons/Tables.png</iconset>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <property name="leftMargin">
    <number>0</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>0</number>
   </property>
   <property name="bottomMargin">
    <number>0</number>
   </property>
   <item>
    <layout class="QHBoxLayout" name="horizontalLayout_3">
     <property name="sizeConstraint">
      <enum>QLayout::SetMinimumSize</enum>
     </property>
     <item>
      <widget class="qMRMLNodeComboBox" name="TableNodeSelector" native="true">
       <property name="nodeTypes" stdset="0">
        <stringlist notr="true">
         <string>vtkMRMLTableNode</string>
        </stringlist>
       </property>
       <property name="renameEnabled" stdset="0">
        <bool>true</bool>
       </property>
      </widget>
     </item>
     <item>
      <widget class="QPushButton" name="LockTableButton">
       <property name="sizePolicy">
        <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
         <horstretch>0</horstretch>
         <verstretch>0</verstretch>
        </sizepolicy>
       </property>
       <property name="minimumSize">
        <size>
         <width>30</width>
         <height>0</height>
        </size>
       </property>
       <property name="toolTip">
        <string>Allow table editing</string>
       </property>
       <property name="text">
        <string/>
       </property>
       <property name="icon">
        <iconset resource="../../../../../Base/QTGUI/Resources/qSlicerBaseQTGUI.qrc">
         <normaloff>:/Icons/Medium/SlicerUnlock.png</normaloff>:/Icons/Medium/SlicerUnlock.png</iconset>
       </property>
       <property name="checkable">
        <bool>false</bool>
       </property>
       <property name="flat">
        <bool>false</bool>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item>
    <widget class="ctkCollapsibleButton" name="DisplayEditCollapsibleWidget" native="true">
     <property name="sizePolicy">
      <sizepolicy hsizetype="Preferred" vsizetype="Expanding">
       <horstretch>0</horstretch>
       <verstretch>1</verstretch>
      </sizepolicy>
     </property>
     <property name="text" stdset="0">
      <string>Edit</string>
     </property>
     <property name="checked" stdset="0">
      <bool>true</bool>
     </property>
     <layout class="QVBoxLayout" name="verticalLayout_8">
      <item>
       <layout class="QHBoxLayout" name="horizontalLayout">
        <property name="spacing">
         <number>3</number>
        </property>
        <item>
         <widget class="QToolButton" name="CopyButton">
          <property name="toolTip">
           <string>Copy</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../../Base/QTGUI/Resources/qSlicerBaseQTGUI.qrc">
            <normaloff>:/Icons/Medium/SlicerEditCopy.png</normaloff>:/Icons/Medium/SlicerEditCopy.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QToolButton" name="PasteButton">
          <property name="toolTip">
           <string>Paste</string>
          </property>
          <property name="text">
           <string>...</string>
          </property>
          <property name="icon">
           <iconset resource="../../../../../Base/QTGUI/Resources/qSlicerBaseQTGUI.qrc">
            <normaloff>:/Icons/Medium/SlicerEditPaste.png</normaloff>:/Icons/Medium/SlicerEditPaste.png</iconset>
          </property>
         </widget>
        </item>
        <item>
         <widget class="QFrame" name="EditControlsFrame">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>15</height>
           </size>
          </property>
          <property name="frameShape">
           <enum>QFrame::NoFrame</enum>
          </property>
          <property name="frameShadow">
           <enum>QFrame::Plain</enum>
          </property>
          <property name="lineWidth">
           <number>0</number>
          </property>
          <layout class="QHBoxLayout" name="horizontalLayout_7">
           <property name="spacing">
            <number>0</number>
           </property>
           <property name="leftMargin">
            <number>0</number>
           </property>
           <property name="topMargin">
            <number>0</number>
           </property>
           <property name="rightMargin">
            <number>0</number>
           </property>
           <property name="bottomMargin">
            <number>0</number>
           </property>
           <item>
            <widget class="QToolButton" name="PlotButton">
             <property name="toolTip">
              <string>Generate an Interactive Plot based on user-selection of the columns of the table. The First (from left to right) Column will be used as X-Axis and each additional Column will be plotted in the same Chart as Y-Axis.</string>
             </property>
             <property name="text">
              <string/>
             </property>
             <property name="icon">
              <iconset resource="../../../../../Base/QTGUI/Resources/qSlicerBaseQTGUI.qrc">
               <normaloff>:/Icons/Medium/SlicerInteractivePlotting.png</normaloff>:/Icons/Medium/SlicerInteractivePlotting.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::MinimumExpanding</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>16</width>
               <height>20</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QToolButton" name="ColumnInsertButton">
             <property name="toolTip">
              <string>Add column</string>
             </property>
             <property name="text">
              <string>...</string>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>:/Icons/column_add.png</normaloff>:/Icons/column_add.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="ColumnDeleteButton">
             <property name="toolTip">
              <string>Delete column</string>
             </property>
             <property name="text">
              <string>...</string>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>:/Icons/column_delete.png</normaloff>:/Icons/column_delete.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="LockFirstColumnButton">
             <property name="toolTip">
              <string>Lock first column</string>
             </property>
             <property name="text">
              <string>...</string>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>:/Icons/row_header_lock.png</normaloff>:/Icons/row_header_lock.png</iconset>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
            </widget>
           </item>
           <item>
            <spacer name="horizontalSpacer_2">
             <property name="orientation">
              <enum>Qt::Horizontal</enum>
             </property>
             <property name="sizeType">
              <enum>QSizePolicy::Preferred</enum>
             </property>
             <property name="sizeHint" stdset="0">
              <size>
               <width>8</width>
               <height>0</height>
              </size>
             </property>
            </spacer>
           </item>
           <item>
            <widget class="QToolButton" name="RowInsertButton">
             <property name="toolTip">
              <string>Add row</string>
             </property>
             <property name="text">
              <string>...</string>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>:/Icons/row_add.png</normaloff>:/Icons/row_add.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="RowDeleteButton">
             <property name="toolTip">
              <string>Delete row</string>
             </property>
             <property name="text">
              <string>...</string>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>:/Icons/row_delete.png</normaloff>:/Icons/row_delete.png</iconset>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QToolButton" name="LockFirstRowButton">
             <property name="toolTip">
              <string>Lock first row</string>
             </property>
             <property name="text">
              <string>...</string>
             </property>
             <property name="icon">
              <iconset>
               <normaloff>:/Icons/column_header_lock.png</normaloff>:/Icons/column_header_lock.png</iconset>
             </property>
             <property name="checkable">
              <bool>true</bool>
             </property>
             <property name="checked">
              <bool>true</bool>
             </property>
            </widget>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </item>
      <item>
       <widget class="qMRMLTableView" name="TableView">
        <property name="sizePolicy">
         <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
          <horstretch>0</horstretch>
          <verstretch>0</verstretch>
         </sizepolicy>
        </property>
        <property name="editTriggers">
         <set>QAbstractItemView::AnyKeyPressed|QAbstractItemView::DoubleClicked|QAbstractItemView::EditKeyPressed|QAbstractItemView::SelectedClicked</set>
        </property>
        <property name="horizontalScrollMode">
         <enum>QAbstractItemView::ScrollPerPixel</enum>
        </property>
       </widget>
      </item>
      <item>
       <widget class="ctkCollapsibleButton" name="SelectedColumnPropertiesCollapsibleButton" native="true">
        <property name="text" stdset="0">
         <string>Column properties</string>
        </property>
        <property name="collapsed" stdset="0">
         <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_2">
         <item>
          <widget class="qSlicerTableColumnPropertiesWidget" name="SelectedColumnPropertiesWidget" native="true"/>
         </item>
        </layout>
       </widget>
      </item>
      <item>
       <widget class="ctkCollapsibleButton" name="NewColumnPropertiesCollapsibleButton" native="true">
        <property name="text" stdset="0">
         <string>New column defaults</string>
        </property>
        <property name="collapsed" stdset="0">
         <bool>true</bool>
        </property>
        <layout class="QVBoxLayout" name="verticalLayout_3">
         <item>
          <widget class="qSlicerTableColumnPropertiesWidget" name="NewColumnPropertiesWidget" native="true">
           <property name="columnNameVisible" stdset="0">
            <bool>false</bool>
           </property>
           <property name="confirmTypeChange" stdset="0">
            <bool>false</bool>
           </property>
          </widget>
         </item>
        </layout>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <spacer name="verticalSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Expanding</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>1</height>
      </size>
     </property>
    </spacer>
   </item>
   <item>
    <spacer name="BottomSpacer">
     <property name="orientation">
      <enum>Qt::Vertical</enum>
     </property>
     <property name="sizeType">
      <enum>QSizePolicy::Fixed</enum>
     </property>
     <property name="sizeHint" stdset="0">
      <size>
       <width>20</width>
       <height>0</height>
      </size>
     </property>
    </spacer>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>qMRMLNodeComboBox</class>
   <extends>QWidget</extends>
   <header>qMRMLNodeComboBox.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qMRMLTableView</class>
   <extends>QTableView</extends>
   <header>qMRMLTableView.h</header>
  </customwidget>
  <customwidget>
   <class>qSlicerWidget</class>
   <extends>QWidget</extends>
   <header>qSlicerWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>qSlicerTableColumnPropertiesWidget</class>
   <extends>qSlicerWidget</extends>
   <header>qSlicerTableColumnPropertiesWidget.h</header>
   <container>1</container>
  </customwidget>
  <customwidget>
   <class>ctkCollapsibleButton</class>
   <extends>QWidget</extends>
   <header>ctkCollapsibleButton.h</header>
   <container>1</container>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../qSlicerTablesModule.qrc"/>
  <include location="../../../../../Base/QTGUI/Resources/qSlicerBaseQTGUI.qrc"/>
 </resources>
 <connections>
  <connection>
   <sender>TableNodeSelector</sender>
   <signal>currentNodeChanged(vtkMRMLNode*)</signal>
   <receiver>TableView</receiver>
   <slot>setMRMLTableNode(vtkMRMLNode*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>240</x>
     <y>10</y>
    </hint>
    <hint type="destinationlabel">
     <x>204</x>
     <y>758</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerTablesModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TableNodeSelector</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>293</x>
     <y>6</y>
    </hint>
    <hint type="destinationlabel">
     <x>389</x>
     <y>20</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>qSlicerTablesModuleWidget</sender>
   <signal>mrmlSceneChanged(vtkMRMLScene*)</signal>
   <receiver>TableView</receiver>
   <slot>setMRMLScene(vtkMRMLScene*)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>175</x>
     <y>179</y>
    </hint>
    <hint type="destinationlabel">
     <x>175</x>
     <y>231</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
